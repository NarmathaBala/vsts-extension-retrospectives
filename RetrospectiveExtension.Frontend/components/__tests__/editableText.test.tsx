import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import EditableText, { EditableTextProps } from '../editableText';

let testText = '';
const mockOnSave = jest.fn(() => { });

const defaultTestProps: EditableTextProps = {
  text: testText,
  isChangeEventRequired: false,
  onSave: mockOnSave
}

describe('Editable Text Component', () => {

  it('renders correctly.', () => {
    const wrapper = shallow(<EditableText {...defaultTestProps} />);
    const component = wrapper.children().dive();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('updates text appropriately.', () => {
    testText = 'Test Text';

    const wrapper = shallow(<EditableText {...defaultTestProps} />);
    const component = wrapper.children().dive();
    expect(toJson(component)).toMatchSnapshot();
  });
});