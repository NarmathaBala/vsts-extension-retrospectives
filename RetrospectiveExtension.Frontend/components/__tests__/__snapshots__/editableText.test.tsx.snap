// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editable Text Component renders correctly. 1`] = `
<div
  aria-live="assertive"
  className="editable-text-container"
>
  <StyledTextFieldBase
    ariaLabel="Please enter feedback title"
    autoAdjustHeight={true}
    autoFocus={true}
    className="editable-text-input-container"
    inputClassName="editable-text-input"
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    resizable={false}
    value=""
  />
</div>
`;

exports[`Editable Text Component updates text appropriately. 1`] = `
<div
  aria-live="assertive"
  className="editable-text-container"
>
  <StyledTextFieldBase
    ariaLabel="Please enter feedback title"
    autoAdjustHeight={true}
    autoFocus={true}
    className="editable-text-input-container"
    inputClassName="editable-text-input"
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    resizable={false}
    value=""
  />
</div>
`;
